tests/Unit/NumericalAlgorithms/LinearSolver/DistributedLinearSolverAlgorithmTestHelpers.hpp

```diff
@@ -121,14 +123,14 @@ struct CollectAp {
   template <
       typename... DbTags, typename... InboxTags, typename Metavariables,
       typename ActionList, typename ParallelComponent,
-      Requires<tmpl2::flat_any_v<cpp17::is_same_v<VariablesTag, DbTags>...>> =
+      Requires<tmpl2::flat_any_v<cpp17::is_same_v<operator_tag, DbTags>...>> =
```

Comment:
Without the `Requires` the lookup in `Parallel::contribute_to_reduction` fails, I think because actions are instantiated also for an empty DataBox but I want to select the instantiation that has the tags. I'm not entirely clear on the mechanism, @nilsdeppe could you confirm/clarify?

Reply:
Yes, though if you only want to eliminate the empty case I suggest you do `Requires<sizeof...(DbTags) != 0>` since that's less work for the compiler.

Reply:
If we want to just exclude the empty case, could you not just replace the `typename... DbTags` with `typename DbTag0, typename... DbTags` and then `DbTags...` with `DbTag0, DbTags...` wherever used in the definition body and avoid the SFINAE altogether?

Reply:
I would suggest just excluding the empty case, yes.

You might want two different cases if you want the action implementation to change depending on what tags are present. One possible use case is doing something like checking if primitive variables exist and then doing something extra.

Reply:
changed and squashed